File: 贪吃蛇_重构/HighestScoreScene.cpp
#include "HighestScoreScene.h"
#include "Console.h"
#include "Config.h"
#include "conio.h"

HighestScoreScene::HighestScoreScene()
{
}

HighestScoreScene::~HighestScoreScene()
{
}

void HighestScoreScene::init()
{
	Console *console = Console::getInstance();
	console->moveCursor(console->getWindowWidth() / 2 - 4, console->getWindowHeight() / 4 + 5);
	printf("最高分：%d", Config::getInstance()->highestScore);
	console->moveCursor(console->getWindowWidth() / 2 - 9, console->getWindowHeight() / 4 + 10);
	printf("按任意键返回主菜单...");
}

int HighestScoreScene::run()
{
	//循环检测按键，如果有按键按下的时候就吸收按键并结束这个场景的运行
	while (true)
	{
		if (_kbhit())
		{
			_getch();
			break;
		}
		Sleep(1);
	}
	return Scene::run();
}

File: snake_cs/Scene/HighestScoreScene.cs
using System;
using System.Threading;

namespace Snake.Scene
{
    /// <summary>
    /// 最高分场景
    /// </summary>
    class HighestScoreScene : Scene
    {
        /// <summary>
        /// 初始化最高分场景
        /// </summary>
        public override void init()
        {
            Console.SetCursorPosition(Console.WindowWidth / 2 - 4, Console.WindowHeight / 3);
            Console.Write("最高分");
            Console.SetCursorPosition(Console.WindowWidth / 2 - 3, Console.WindowHeight / 3 + 4);
            Console.Write("{0}分", Config.HighestScore);
            Console.SetCursorPosition(Console.WindowWidth / 2 - 11, Console.WindowHeight / 3 + 8);
            Console.Write("按任意键回到开始界面...");
        }

        /// <summary>
        /// 运行最高分场景
        /// </summary>
        /// <returns>运行相关信息</returns>
        public override int run()
        {
            while (true)
            {
                if (Console.KeyAvailable)
                {
                    Console.ReadKey(true);
                    break;
                }
                Thread.Sleep(1);
            }
            return base.run();
        }
    }
}
